### PR TITLE
Bump go.opentelemetry.io/otel to 0.16.0 in /examples/demo/app

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: examples/demo/app/main.go [hunk 1]
```diff
@@ -48,12 +49,12 @@ func initProvider() func() {
...
-		otlp.WithAddress(otelAgentAddr),
-		otlp.WithGRPCDialOption(grpc.WithBlock()), // useful for testing
-	)
-	handleErr(err, "failed to create exporter")
+	exp, err := otlp.NewExporter(ctx, otlpgrpc.NewDriver(
+		otlpgrpc.WithInsecure(),
+		otlpgrpc.WithEndpoint(otelAgentAddr),
+		otlpgrpc.WithDialOption(grpc.WithBlock()), // useful for testing
+	))
+	handleErr(err, "failed to create resource")
```
Comment (tigrannajaryan) [inline examples/demo/app/main.go:57]: Is this change intentional? It appears that the previous line is exporter creation, not resource creation.

**Review by tigrannajaryan — APPROVED** [review]

File: examples/demo/app/main.go (hunk 1 above)
Comment (bogdandrutu) [reply examples/demo/app/main.go]: Good catch. Fixed
